nebula/ui/components/MZMenu.qml

```diff
@@ -103,10 +103,11 @@ Item {
 
         sourceComponent: menuBar.rightButtonComponent
         onItemChanged: {
-            if (item instanceof Text) {
+            //item.item is for right button components that might be inside a loder itself
+            if (item instanceof Text || item.item instanceof Text) {
```

Comment:
For the item.item condition, should the first item be checked if it is an instance of Loader? Also, are any null checks required?

Reply:
Hmm, null checks for what? `item`? If `item` is null, `item instanceof Text` will evaluate to `false`

Reply:
Please ignore the null check in the comment.